libs/cardano-ledger-core/src/Cardano/Ledger/Tools.hs

```diff
@@ -137,6 +139,9 @@ estimateMinFeeTx pp tx numKeyWits numByronKeyWits = setMinFeeTx pp tx' ^. bodyTx
           , Byron.aaNetworkMagic = Byron.NetworkTestnet maxBound
           }
 
+integerToByteStringN :: Int -> Integer -> (ByteString, Maybe Integer)
+integerToByteStringN len = BS.unfoldrN len (\n -> Just (fromInteger n, n `shiftR` 8))
```

Comment:
Good idea on extracting this into a utility function. Although, I doubt that left over `Integer` will be useful. Also we can make it more general, it doesn't have to be `Integer`.
```suggestion
integralToByteStringN :: (Integral i, Bits i) => Int -> i -> ByteString
integralToByteStringN len = fst . BS.unfoldrN len (\n -> Just (fromIntegral n, n `shiftR` 8))
```